scripts/fetch_data.py

```diff
@@ -0,0 +1,274 @@
+import csv
+from statistics import mean
+import numpy as np
+import logging
+
+from ce.api.models import models
+from ce.api.multistats import multistats
+
+
+logger = logging.getLogger('scripts')
+
+
+def get_dict_val(dict, val):
+    """Given a dictionary key name return the associated value"""
+    try:
+        return dict[val]
+    except KeyError as e:
+        logger.warning('Unable to get val: {} from dict: {} error: {}'
+                       .format(val, dict, e))
+        return None
+
+
+def read_csv(filename):
+    """Read a csv file which contains at least the id and condition columns
+       and return those columns applying a 'rule_' prefix to the id column.
+    """
+    rules = {}
+    with open(filename, 'r') as f:
+        csv_reader = list(csv.DictReader(f, delimiter=';'))
+        rules = {'rule_{}'.format(list(row.values())[0]): list(row.values())[1]
+                 for row in csv_reader}
+
+    return rules
+
+
+def filter_by_period(target, dates, periods):
+    """Search through dictionary containing data for different 30 year periods,
+       find the desired period and return the target variable.
+    """
+    for key in periods.keys():
+        for date in dates:
+            if date in key:
+                try:
+                    return periods[key][target]
+                except KeyError as e:
+                    logger.warning('Bad target variable: {} error: {}'
+                                   .format(target, e))
+                    return None
```

Comment:
Seems like you're just masking the problem here, without doing anything useful. Again, should this be a warning or an outright error? Isn't `None` just going to propagate and cause (stranger) problems further down the line of execution?